### ISSUE TITLE
Human in the Loop Implementation needs

### ISSUE THREAD

**Issue by ClaraCogswell** [hydroshare/hydroshare#4726]
Revisit end of August 2022:
Human in the loop is a much requested feature that will implement the following process:

The publisher will submit their resource for publication. A popup will appear that displays the following information:
      Metadata Minimum
      Required:
      Title: 
      minimum 30 characters
      
      Abstract: 
      Proposed: minimum 150 characters. 
      
      Subject Keywords:
      Minimum 3 subject keywords, including geographic location keywords
      
      Recommended but Not Required:
      Funding Agency:
      At least 1 funding agency
      
      Readme File
      All variables, abbreviations/acronyms, and non-standard file formats should be defined in the ReadMe file. If folder structure 
      within the resource is complicated, it can be described in the ReadMe file as well. 
      Geographic Coverage
      Coverage that describes locations that are related to the dataset in whatever capacity the depositor finds most relevant. 

After the user acknowledges the minimum metadata standard (selecting ‘okay’ or something similar) and clicks through the publication agreement, a notification email will be sent to the user that the resource is under review for appropriate minimum metadata and to ensure that it adheres to community guidelines, also that the review process will likely be complete within 1 business day, but not exceed 2 business days. 
If the resource meets the requirements and the content adheres to community guidelines, the reviewer will approve it and the DOI process will be triggered, and the publisher will be notified that their DOI is being generated.

A rejection will trigger an email to the publisher that will explain what is deficient about the resource. The reviewer will write this email informing the publisher what they must do to have their resource approved. The resource will be set back to public while these changes are made, and the publisher will then resubmit their resource for publication. 

**Implementation:**
Needed Features:
- [ ] The “Minimum Metadata for published resources” pop-up when the publish icon is pressed.
- [ ] The “Publication Pending” status for the resource in review
- [ ] The email to the publisher that the resource is under review for minimum metadata and adherence to community guidelines
- [ ] The email to the reviewer at [help@cuahsi.org](mailto:help@cuahsi.org) to review the resource
- [ ] The ability to accept the resource and trigger the DOI process
- [ ] The ability to reject the resource and open an email window to email the publisher with the explanation
- [ ] The action of setting the resource back to ‘public’ when it is rejected 

Clara will need to:

- [ ] Create a new help section on the minimum metadata requirement under the publishing section
- [ ] Write a newsletter article about the implementation so that users know it is coming

**Comment by dtarb** [comment]
This updates previous work on this in outlined in #2298 and addresses [policy for checking resources prior to publication](https://docs.google.com/document/d/1TEWjijx-ZR6NR_Zy-1GIOjy9ZqGkFb64cE_2d-OevFY/edit?usp=sharing).

**Comment by devincowan** [comment]
@ClaraCogswell you have 

> The email to the publisher that the resource is under review for minimum metadata and adherence to community guidelines

as a feature. Does this have to be an email? What if we simply show this info in the HS UI once they submit the publication request? It would be more direct feedback and avoids extra emails.

Also, I'm thinking that it might be nice to include the relevant metadata sections in the email to the reviewer. So instead of navigating to the resource within HS, the reviewer could read the relevant sections directly in the email and then select (in email) "accept" or "reject". Do you see a problem with that approach?

**Comment by devincowan** [comment]
yes, my thought was to include the abstract, keywords, etc in the email to the reviewer

**Comment by ClaraCogswell** [comment]
That sounds like it would be fine to me, you're right it would streamline the process.

**Comment by devincowan** [comment]
@martinseul and I talked this morning -- decided that adding yet another popup (there are already 2 that show up when one decides to publish) is poor form. So plan to provide the min metadata list on the relevant help page.

**Comment by ClaraCogswell** [comment]
You know best what good form and bad form would be so I'll leave it to you. The min metadata will be added to the help pages related to publishing, and cross listed in a few places.

**Comment by horsburgh** [comment]
I think that the same rules for metadata editing should apply for published resources and those that have been submitted for publication, but prior to a DOI being granted.

The reviewer should look at the landing page for the resource to complete the review. So, the acceptance or rejection of the resource should be based on whatever the reviewer sees at the time of the review. 

The reality is that we allow editing of published resources, so there is nothing stopping users from modifying Abstract, Keywords, Funding Agency, and Coverage information even after the review takes place. The readme file is part of the content files, so it should be immutable upon submission for publication.

**Comment by devincowan** [comment]
Good point, thanks @horsburgh.
